Code/GraphMol/MolOps.cpp

```diff
@@ -152,6 +152,43 @@ void phosphorusCleanup(RWMol &mol, Atom *atom) {
   // force a recalculation of the explicit valence here
   atom->calcExplicitValence(false);
 }
+
+void halogenCleanup(RWMol &mol, Atom *atom) {
+  PRECONDITION(atom, "bad atom");
+  // Conversions done:
+  //    X(=O)(=O)(=O)O -> [X+3]([O-])([O-])([O-])O
+  //    X(=O)(=O)O -> [X+2]([O-])([O-])O
+  //    X(=O)O -> [X+]([O-])O
+  int ev = atom->calcExplicitValence(false);
```

Comment:
Broken record: prefer named constants so I don't have to look up calcExplicitValence :)

Perhaps github needs a code-review API lookup feature!  You'd make tens of dollars if you wrote that.